apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -184,6 +193,43 @@ verify(Hash, Nonce, Soln) ->
             {error, {unknown, {C, E}}}
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc
+%%   Creates the 80-byte Cuckoo buffer (hex encoded) from a base64-encoded
+%%   hash and a uint64 nonce.
+%% @end
+%%------------------------------------------------------------------------------
+-spec pack_header_and_nonce(binary(), integer()) -> string().
+pack_header_and_nonce(Hash, Nonce) ->
+    %% Cuckoo originally uses 32-bit nonces inserted at the end of its 80-byte buffer.
+    %% This buffer is hashed into the keys used by the main algorithm.
+    %%
+    %% We insert our base64-encoded 64-bit Nonce before that (i.e., from byte 65)
+    %% and pass 0 as nonce  for cuckoo. (We use the -x option to write the full buffer.)
+    %% The SHA256 hash is 32 bytes, 44 bytes base64-encoded. There is still room for
+    %% zero characters between header and nonce.
+    %%
+    %% (Base64 encoding: see RFC 3548, Section 3:
+    %% https://tools.ietf.org/html/rfc3548#page-4
+    %% converts every triplet of bytes to 4 characters: from N bytes to 4*ceil(N/3)
+    %% bytes.)
+    NonceStr = base64:encode_to_string(<<Nonce:64/little-unsigned-integer>>),
+    NonceStart = ?CUCKOO_BUFSIZE - length(NonceStr) - 4,
+    HashStr = base64:encode_to_string(Hash),
+    case length(HashStr) of
```

Comment:
Please remove hardcode expectation at this point. I.e. no case as there is actually no conditional I understand - only one code path.
(Feel free to perform some refactoring for clarifying code if you feel that appropriate.)